### PR TITLE
 Hippie self administration and automatic grief prevention system part 1. 

### PR DESCRIPTION
**Pull request by GuyonBroadway**
:cl: GuyonBroadway
admin: Implemented a special self administration system that will effectively make griefing impossible, this system is fully tested and foolproof.
del: Ahelping is partially disabled as its no longer necessary, the verb has been kept around for logging and legacy purposes. 
/:cl:

[why]: Auto-moderation is the future, this will give our current admins a much needed leg-up what with not having to administrate as hard.

### CODE REVIEW

**Comment by Pyko1** [comment]
anything that helps us to lessen teh load on admins is welcome

**Comment by GuyonBroadway** [comment]
I've got some big designs for part 2, automatic token based banning, a simple Neural-net to identify and flag potential shitlers and a shared database of known griefers that exists between servers. Hopefully we can get MSO on board.

**Comment by Pyko1** [comment]
we need to also add the proxy detection and account age as a factor for grief detection

File: hippiestation/code/datums/traits/negative.dm [hunk 1]
```diff
@@ -45,3 +45,20 @@
...
+	desc = "A foolproof system that will eiliminate any and all grief, while it's listed here, this is non-optional."
+	value = 0
+	mob_trait = TRAIT_PACIFISM
+	gain_text = "<span class='danger'>You feel repulsed by the thought of griefing!</span>"
+	lose_text = "<span class='notice'>Error, Anti-Griff offline, contact an admin.</span>"
+	medical_record_text = "Patient is unusually pacifistic and cannot bring themselves to cause physical harm."
+
+/datum/trait/nonviolent/on_process()
+	if(trait_holder.mind && LAZYLEN(trait_holder.mind.antag_datums))
+		to_chat(trait_holder, "<span class='boldannounce'>Your antagonistic nature allows you to buypass the otherwise foolproof hippie anti-griff system.</span>")
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/negative.dm]: "bypass"

Comment (JohnGinnane) [reply hippiestation/code/datums/traits/negative.dm]: "Hippie"

Comment (ghost) [reply hippiestation/code/datums/traits/negative.dm]: make it so you can pay antag tokens to remove it then

File: hippiestation/code/datums/traits/negative.dm [hunk 2]
```diff
@@ -45,3 +45,20 @@
 		return
 	else
 		B.inv.storage_slots = initial(B.inv.storage_slots)
+
+/datum/trait/nonviolent
+	name = "Hippie Anti-Griff system."
```
Comment (JohnGinnane) [inline hippiestation/code/datums/traits/negative.dm]: remove "."

Comment (JohnGinnane) [reply hippiestation/code/datums/traits/negative.dm]: probably capitalise "system"

**Comment by Jujumatic** [comment]
Eliminate? :D

**Comment by GuyonBroadway** [comment]
closing this pr to make some fixes, I'll open it back up when its ready.

**Comment by Pyko1** [comment]
Not sure, if we actually should add play_time based restrictions on jobs as well. it would help at least against people bypassing bans.

**Comment by GuyonBroadway** [comment]
Something else I need to consider is cargo ordering implants, maybe if I make the lockboxes have armour to resist fire/acid/fireaxe then only a head of staff can open it.

There is the issue of someone stealing a Head of Staff ID but the rest of the system should prevent most cases of that.

**Comment by GuyonBroadway** [comment]
Just need to pop this open and closed so it picks up my last push.

**Comment by GuyonBroadway** [comment]
Ok that seems to have done it, closed pending further changes.

**Pull request merged**